ouroboros-network-api/src/Ouroboros/Network/BlockFetch/ConsensusInterface.hs

```diff
@@ -199,6 +199,8 @@ data ChainComparison header =
        -- This is used as part of selecting which chains to prioritise for
        -- downloading block bodies.
        --
+       -- PRECONDITION: The two fragments must intersect.
```

Comment:
An alternative here would be to modify `compareCandidateChains` to also take the
current selection as an argument and require that the other two fragments
intersect with it (but don't necessarily intersect themselves). This would avoid modifying `ChainSuffix` in the rest of this commit.